map/autoware_lanelet2_map_validator/src/include/lanelet2_map_validator/utils.hpp

```diff
@@ -27,10 +28,16 @@ namespace autoware
 {
 namespace validation
 {
-template <typename T>
-void appendIssues(std::vector<T> & to, std::vector<T> && from)
+template <typename Container, typename T>
+auto appendIssues(std::vector<T> & to, Container && from) ->
+  typename std::enable_if_t<std::is_same_v<T, typename std::decay_t<Container>::value_type>, void>
 {
-  to.insert(to.end(), std::make_move_iterator(from.begin()), std::make_move_iterator(from.end()));
+  if constexpr (std::is_rvalue_reference<decltype(from)>::value) {
```

Comment:
I fixed `appendIssue` to take universal reference of `from` to optimize the copy.

The trailing return type denotes the type constraint (concept) that `T` and `std::vector<T>::value_type` matches